modules/widget/list/templates/list.widget.js.tmpl

```diff
@@ -559,7 +559,7 @@ Widget.types.list = {
 	/**
 	 * Reloads the list. Must be called with true for load_heading on the first load.
 	 */
-	reload: function(load_heading) {
+	reload: function(load_heading, keep_paging) {
```

Comment:
This additional param fixes an issue that was always present but hadn’t been as prominent until now: when list is reloaded by a third-party, the paging was not reset. But since previously paging didn’t influence list size, it wasn’t as noticeable.